### PR TITLE
HNI-Scantrad - move to individual

### LINKED ISSUE #1136: [HNI-Scantrad] Requesting updated URL Update
### Source information

HNI-Scantrad 1.4.4

### Source language

English

### Source new URL

https://hni-scantrad.net/comics/hajime-no-ippo

### Other details

It appears they have only uploaded chapters 1346 to the latest to this new site.

### Acknowledgements

- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL has not changed yet.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by SnakeDoc83**
Closes #1136

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/all/hniscantrad/src/eu/kanade/tachiyomi/extension/all/hniscantrad/HniScantrad.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        throw UnsupportedOperationException()
+    }
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$baseUrl/api/search/$query", headers)
```
Comment (AwkwardPeak7) [inline src/all/hniscantrad/src/eu/kanade/tachiyomi/extension/all/hniscantrad/HniScantrad.kt:76]: should use HttpUrl builder so query is url encoded

File: src/all/hniscantrad/src/eu/kanade/tachiyomi/extension/all/hniscantrad/HniScantrad.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    override val baseUrl = "https://hni-scantrad.net"
+
+    override val name = "HNI-Scantrad"
+
+    override val supportsLatest = false
+
+    private val json by injectLazy<Json>()
+
+    private fun JsonElement?.getContent() = this?.jsonPrimitive?.content ?: ""
```
Comment (AwkwardPeak7) [inline src/all/hniscantrad/src/eu/kanade/tachiyomi/extension/all/hniscantrad/HniScantrad.kt:36]: use Serializable classes, it's more pretty

**Comment by Hinalover** [comment]
Ahk!! I thought this was just a URL change, not a complete redesign!

**Comment by stevenyomi** [comment]
Uses PizzaReader: https://github.com/FedericoHeichou/PizzaReader

**Pull request closed without merging**